python-stdlib/shutil/shutil.py

```diff
@@ -27,3 +28,14 @@ def copyfileobj(src, dest, length=512):
             if not buf:
                 break
             dest.write(buf)
+            
+def disk_usage(path):
+    _ntuple_diskusage = namedtuple('usage', 'total used free')
```

Comment:
This will recreate the namedtuple every time this function is called, which is not really efficient.

You could create it at the global scope unconditionally.  Or make `_ntuple_diskusage` global and start out as `None`, then populate it with the namedtuple the first time this function is called.

Also, better to use `namedtuple('usage', ('total', 'used', 'free'))`, it's more compatibly with MicroPython ports (that don't enable `MICROPY_CPYTHON_COMPAT`) and more efficient when the code is frozen.

Reply:
Thanks @dpgeorge the latest commit should resolve both of those issues.